index.js

```diff
@@ -53,22 +54,11 @@ function checkFragments(ast) {
         fragmentSourceMap[fragmentName] = {};
         fragmentSourceMap[fragmentName][sourceKey] = true;
       }
-
-      if (!astFragmentMap[sourceKey]) {
-        astFragmentMap[sourceKey] = true;
-        return true;
-      } else {
-        // This fragmentDefinition already exists in the AST, no need to add it again.
-        return false;
-      }
-    } else {
-      // This definition is not a fragment.
-      return true;
     }
-  });
+  }
 
   return assign({}, ast, {
-    definitions: uniqueFragmentDefinitions
+    definitions: uniqBy(ast.definitions, (definition) => cacheKeyFromLoc(definition.loc))
```

Comment:
interestingly - since we're not compiling any of this code, this function declaration actually got stuck in our uglify process. something to note when making changes to this repo / code in the future that whatever we have here is exactly what's going into browsers